clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/RestoreSnapshotAtomicOperation.groovy

```diff
@@ -0,0 +1,165 @@
+
+import com.netflix.spinnaker.clouddriver.core.services.Front50Service
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.google.deploy.description.SerializeApplicationDescription.RestoreSnapshotDescription
+import com.netflix.spinnaker.clouddriver.google.model.GoogleCluster
+import com.netflix.spinnaker.clouddriver.google.model.GoogleSecurityGroup
+import com.netflix.spinnaker.clouddriver.google.model.GoogleServerGroup
+import com.netflix.spinnaker.clouddriver.google.model.loadbalancing.GoogleLoadBalancerView
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleClusterProvider
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleLoadBalancerProvider
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleSecurityGroupProvider
+import com.netflix.spinnaker.clouddriver.google.security.GoogleNamedAccountCredentials
+import com.netflix.spinnaker.clouddriver.jobs.JobExecutor
+import com.netflix.spinnaker.clouddriver.jobs.JobRequest
+import com.netflix.spinnaker.clouddriver.jobs.JobStatus
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import org.springframework.beans.factory.annotation.Autowired
+
+
+class RestoreSnapshotAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "RESTORE_SNAPSHOT"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  private final RestoreSnapshotDescription description
+  private final String applicationName
+  private final String accountName
+  private final Long snapshotTimestamp
+  private String project
+  private List applicationTags
+
+  @Autowired
+  GoogleClusterProvider googleClusterProvider
+
+  @Autowired
+  GoogleLoadBalancerProvider googleLoadBalancerProvider
+
+  @Autowired
+  GoogleSecurityGroupProvider googleSecurityGroupProvider
+
+  @Autowired
+  AccountCredentialsRepository accountCredentialsRepository
+
+  @Autowired
+  JobExecutor jobExecutor
+
+  RestoreSnapshotAtomicOperation(RestoreSnapshotDescription description) {
+    this.description = description
+    this.applicationName = description.applicationName
+    this.accountName = description.accountName
+    this.snapshotTimestamp = description.snapshotTimestamp
+    this.applicationTags = []
+  }
+
+  /* curl -X POST -H "Content-Type: application/json" -d '[ { "restoreSnapshot": { "applicationName": "example", "credentials": "my-google-account" }} ]' localhost:7002/gce/ops */
+  @Override
+  Void operate(List priorOutputs) {
+    def credentials = accountCredentialsRepository.getOne(this.accountName) as GoogleNamedAccountCredentials
+    this.project = credentials.project
+
+    // Make directory for terraform files
+    def makeDirCommand = ["mkdir", "$applicationName-$accountName"]
```

Comment:
Would prefer to use the java-provided mechanism for creating directories as opposed to relying on the job runner mechanism.
Same for deleting the directory, even though it will require a bit of work to first empty the directory.